### PR TITLE
TextEditor: improve performance of initial rendering

### PR DESCRIPTION
**Pull request by EugeniyKiyashko**

### CODE REVIEW

**Review by iBat — DISMISSED** [review]

**Review by alexanderPolosatov — DISMISSED** [review]

File: packages/devextreme/js/ui/text_box/ui.text_editor.label.js [hunk 1]
```diff
@@ -139,9 +139,11 @@ class TextEditorLabel {
     }
 
     _updateBeforeWidth() {
-        this._$before.css({ width: this._props.beforeWidth });
+        if(this._isVisible()) {
+            this._$before.css({ width: this._props.beforeWidth ?? this._props.getBeforeWidth() });
```
Comment (ksercs) [inline packages/devextreme/js/ui/text_box/ui.text_editor.label.js]: let's move it to a variable

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/textEditorParts/common.tests.js [hunk 2]
```diff
@@ -562,16 +564,53 @@ QUnit.module('label integration', {
...
             this.init();
             const beforeButtonsContainerWidth = getWidth($(`.${BUTTONS_CONTAINER_CLASS}`));
 
-            assert.strictEqual(this.getProps().beforeWidth, beforeButtonsContainerWidth);
+            assert.strictEqual(this.getProps().getBeforeWidth(), beforeButtonsContainerWidth);
+        });
+
+        QUnit.test('editor should not initiate taking of width for label if labelMode is hidden (T1192938)', function(assert) {
+            const originalWidth = implementationsMap.getWidth;
+
```
Comment (ksercs) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/textEditorParts/common.tests.js:580]:
```suggestion
```

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/textEditorParts/common.tests.js [hunk 3]
```diff
@@ -562,16 +564,53 @@ QUnit.module('label integration', {
...
+                implementationsMap.getWidth = originalWidth;
+            }
+        });
+
+        LABEL_MODES.forEach((labelMode) => {
+            QUnit.test(`editor initiate taking of width for label if labelMode is ${labelMode} (T1192938)`, function(assert) {
+                const originalWidth = implementationsMap.getWidth;
+
+                try {
+
```
Comment (ksercs) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/textEditorParts/common.tests.js]:
```suggestion
```

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/textEditorParts/common.tests.js [hunk 4]
```diff
@@ -562,16 +564,53 @@ QUnit.module('label integration', {
...
         QUnit.test('editor should pass beforeWidth equal to before buttons container width', function(assert) {
             this.init();
             const beforeButtonsContainerWidth = getWidth($(`.${BUTTONS_CONTAINER_CLASS}`));
 
-            assert.strictEqual(this.getProps().beforeWidth, beforeButtonsContainerWidth);
+            assert.strictEqual(this.getProps().getBeforeWidth(), beforeButtonsContainerWidth);
+        });
+
+        QUnit.test('editor should not initiate taking of width for label if labelMode is hidden (T1192938)', function(assert) {
+            const originalWidth = implementationsMap.getWidth;
```
Comment (ksercs) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/textEditorParts/common.tests.js]:
```suggestion
            const originalGetWidth = implementationsMap.getWidth;
```

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/textEditorParts/common.tests.js [hunk 5]
```diff
@@ -562,16 +564,53 @@ QUnit.module('label integration', {
...
+
+                assert.strictEqual(implementationsMap.getWidth.callCount, 0);
+            } finally {
+                implementationsMap.getWidth = originalWidth;
+            }
+        });
+
+        LABEL_MODES.forEach((labelMode) => {
+            QUnit.test(`editor initiate taking of width for label if labelMode is ${labelMode} (T1192938)`, function(assert) {
+                const originalWidth = implementationsMap.getWidth;
```
Comment (ksercs) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/textEditorParts/common.tests.js]:
```suggestion
                const originalGetWidth = implementationsMap.getWidth;
```

File: packages/devextreme/js/ui/text_box/ui.text_editor.label.js [hunk 6]
```diff
@@ -157,7 +159,9 @@ class TextEditorLabel {
     }
 
     _updateMaxWidth() {
-        this._$label.css({ maxWidth: this._props.containerWidth });
+        if(this._isVisible() && !this._isOutsideMode()) {
+            this._$label.css({ maxWidth: this._props.containerWidth ?? this._props.getContainerWidth() });
```
Comment (ksercs) [inline packages/devextreme/js/ui/text_box/ui.text_editor.label.js]: let's move it to a variable

File: packages/devextreme/js/ui/text_box/ui.text_editor.label.js (hunk 1 above)
Comment (EugeniyKiyashko) [reply packages/devextreme/js/ui/text_box/ui.text_editor.label.js]: https://github.com/DevExpress/DevExtreme/pull/26247/commits/5322af04efe33ef817500989e942071703dbcb28

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/textEditorParts/common.tests.js (hunk 2 above)
Comment (EugeniyKiyashko) [reply packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/textEditorParts/common.tests.js:580]: https://github.com/DevExpress/DevExtreme/pull/26247/commits/5322af04efe33ef817500989e942071703dbcb28

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/textEditorParts/common.tests.js (hunk 4 above)
Comment (EugeniyKiyashko) [reply packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/textEditorParts/common.tests.js]: https://github.com/DevExpress/DevExtreme/pull/26247/commits/5322af04efe33ef817500989e942071703dbcb28

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/textEditorParts/common.tests.js (hunk 5 above)
Comment (EugeniyKiyashko) [reply packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/textEditorParts/common.tests.js]: https://github.com/DevExpress/DevExtreme/pull/26247/commits/5322af04efe33ef817500989e942071703dbcb28

File: packages/devextreme/js/ui/text_box/ui.text_editor.label.js (hunk 6 above)
Comment (EugeniyKiyashko) [reply packages/devextreme/js/ui/text_box/ui.text_editor.label.js]: https://github.com/DevExpress/DevExtreme/pull/26247/commits/5322af04efe33ef817500989e942071703dbcb28

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/textEditorParts/common.tests.js (hunk 3 above)
Comment (EugeniyKiyashko) [reply packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/textEditorParts/common.tests.js]: https://github.com/DevExpress/DevExtreme/pull/26247/commits/5322af04efe33ef817500989e942071703dbcb28

**Review by alexanderPolosatov — APPROVED** [review]

**Review by iBat — APPROVED** [review]

**Review by ksercs — APPROVED** [review]

**Review by Zedwag — APPROVED** [review]

**Pull request merged**